xpc-connection/tests/message_round_trip.rs

```diff
@@ -0,0 +1,178 @@
+use futures::executor::block_on;
+use futures::StreamExt;
+use std::collections::HashMap;
+use xpc_connection::{Message, XpcConnection};
+
+/// FIXME: Outgoing strings, both as dictionary keys and as string values, are
+/// expected to be explicitly NULL terminated, but incoming strings do not
+/// contain the NULL. This prevents us from being able to compare outgoing and
+/// incoming container dictionaries directly.
+/// This will also make writing an XPC listener a little more confusing as
+/// messages have two different behaviours depending on the context.
+
+#[test]
+#[ignore = "This test requires the echo server to be running"]
+fn send_and_receive_int64() {
```

Comment:
We could make tests `async` (using [`tokio::test`](https://docs.rs/tokio/1.0.2/tokio/attr.test.html) or [`async_std::test`](https://docs.rs/async-std/1.9.0/async_std/attr.test.html)) and return `Result` to make some of the test code more idiomatic.

Reply:
These tests only need a single-threaded executor, and only use one future each, is it worth the extra dev dependency for tokio/async-std?

I wasn't aware that using Result in tests is more idiomatic, can you show some examples? I checked the futures-rs and cargo source code, they use panic and unwrap in tests.

Reply:
Sorry, I hope I didn't offend. I meant making the code in the tests look more idiomatic, not necessarily the tests themselves. I say this because tests act as examples for others and having code that looks the same as you'd normally write helps with that.

As for adding another dependency for tests, I am flexible whatever we choose. Having an extra dependency doesn't bother me much.

Reply:
No offence taken at all 😄, I just thought I might have been missing some info, thanks for clearing it up.

I'm working on some examples as part of the XpcListener branch. I'm hoping to have a privileged daemon, non-privileged daemon (user agent) and a client that can work in various modes (client can become the server, client can connect to another client, etc.).